src/lib389/lib389/__init__.py

```diff
@@ -1730,6 +1730,17 @@ def with_systemd(self):
             return self.systemd_override
         return self.ds_paths.with_systemd
 
+    def with_systemd_running(self):
+        if not self.with_systemd():
+            return False
+        cp = subprocess.run(["systemctl", "is-system-running"],
+                            capture_output=True)
+        # is-system-running can detect the 7 modes (initializing, starting,
+        # running, degraded, maintenance, stopping, offline) or "unknown".
+        # To keep things simple, we assume that anything other than "offline"
+        # means that systemd is usable.
+        return cp.stdout != b'offline\n'
```

Comment:
Thanks for the suggestion. I used `text=True` because the docs say that `universal_newlines` is deprecated.
The newline is still there even with `text=True`, but is that a problem? Systemd doesn't work on any non-unix system, so the newline will always be `\n`. I think `text=True` may also convert the newline if used on OSes where newline is not `\n`, but it doesn't impact us at all. Right?

Reply:
`text` was added in Python 3.7, but we still have to support RHEL 8 with Python 3.6. So even though `universal_newlines` is marked as deprecated, it's still available and not removed yet.
We can use `cp.stdout.strip()` to compare without new lines, this is more about code readability than platform support.

Reply:
That makes sense. I updated the PR again.